hkmc2/shared/src/main/scala/hkmc2/codegen/js/JSBuilder.scala

```diff
@@ -302,17 +304,12 @@ class JSBuilder(using TL, State, Ctx) extends CodeBuilder:
                     doc" # ${mtdPrefix}get ${td.sym.nme}() ${ braced(body(bod, endSemi = true)) }"
                 .mkDocument(" ")
               }${
-                if mtds.exists(_.sym.nme == "toString")
```

Comment:
I thing we should still generate a `toString` method if one isn't set, and that method should call the rendering method in Rendering.

Reply:
The `Rendering` module calls `toString` and use it output as the result if it is defined, which has higher precedence than checking `definitionMetadata`. 🤔 Hence, the change will cause stack overflow.

Reply:
The problem is that we can’t tell whether a `toString` is an automatically generated version that calls the `render` function, or a user-defined version. Hence, we should not call its `toString` in `render` at all, otherwise it might cause stack overflow because two `render` calls don’t share the `visitedObjects` map.

> Can't we pass a flag to the render function that says not to call `toString`?

No, in this case, it should not be called by default.

> The reason we want to define `toString` is to make MLscript definitions "feel" like they work consistently, incl. when inserted into string interpolators etc.

I agree. We can just don’t call `toString` in `Rendering` module, as it also hinders the print width limitation in pretty-printing.

Reply:
Ok, not calling `toString` when rendering seems to be the more reasonable approach for now.